pkg/oc/cli/cmd/rollout/latest.go

```diff
@@ -126,18 +142,73 @@ func (o *RolloutLatestOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command
 
 func (o RolloutLatestOptions) Validate() error {
 	if len(o.infos) != 1 {
-		return errors.New("a deployment config name is required.")
+		return errors.New("a deployment config name is required")
 	}
 	return nil
 }
 
-func (o RolloutLatestOptions) RunRolloutLatest() error {
-	info := o.infos[0]
-	config, ok := info.Object.(*deployapi.DeploymentConfig)
+// rolloutDeployments rollouts the latest version for deployments.
+func (o RolloutLatestOptions) rolloutDeployment(d *kextensions.Deployment) error {
+	out, ok := d.GetAnnotations()[triggerapi.TriggerAnnotationKey]
 	if !ok {
-		return fmt.Errorf("%s is not a deployment config", info.Name)
+		return fmt.Errorf("deployment %s does not have any triggers defined", d.Name)
+	}
+	triggers := []triggerapi.ObjectFieldTrigger{}
+	if err := json.Unmarshal([]byte(out), &triggers); err != nil {
+		return err
+	}
+	var (
+		errors                []error
+		hasPausedImageTrigger bool
+		hasUpdatedImages      bool
+	)
+	obj, err := kapi.Scheme.DeepCopy(d)
+	if err != nil {
+		return err
 	}
+	dCopy := obj.(*kextensions.Deployment)
+	for _, trigger := range triggers {
+		if !trigger.Paused {
+			continue
+		}
+		container, remainder, err := annotations.ContainerForObjectFieldPath(dCopy, trigger.FieldPath)
+		if err != nil || remainder != "image" {
+			continue
+		}
+		hasPausedImageTrigger = true
+		spec, err := triggerutil.ResolveLatestTriggerImagePullSpec(o.imageClient, dCopy.Namespace, trigger.From)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("unable to resolve image %s for container %q: %v", trigger.From.Name, container.GetName(), err))
+			continue
+		}
+		if container.GetImage() != spec.String() {
+			container.SetImage(spec.String())
+			hasUpdatedImages = true
+		}
+	}
+	if !hasPausedImageTrigger {
+		return fmt.Errorf("deployment %q has no paused triggers", dCopy.Name)
+	}
+	if !hasUpdatedImages {
+		return fmt.Errorf("deployment %q already runs the latest images", dCopy.Name)
```

Comment:
I don't think this should be an error. You are asking it to rollout latest and you did so by NOOP. The target state is where user asked it to be.

Reply:
@tnozicka user assumption was that oc rollout latest will cause a new deployment, it hasn't, so for user that is an error. Imagine you writing a script, you can use this to check whether you run the latest image or not (actually that reminds me we should have --dry-run for oc rollout latest ;-)